darts/timeseries.py

```diff
@@ -649,97 +651,112 @@ def from_dataframe(
                 # Try to convert to integers if needed
                 try:
-                    time_col_vals = time_col_vals.astype(int)
-                except ValueError:
+                    time_col_vals = time_col_vals.cast(nw.Int64)
+                except Exception:
                     pass
 
-            if np.issubdtype(time_col_vals.dtype, np.integer):
-                # We have to check all integers appear only once to have a valid index
-                raise_if(
-                    time_col_vals.duplicated().any(),
-                    "The provided integer time index column contains duplicate values.",
-                )
-
+            if time_col_vals.dtype.is_integer():
+                if time_col_vals.is_duplicated().any():
+                    raise_log(
+                        ValueError(
+                            "The provided integer time index column contains duplicate values."
+                        )
+                    )
                 # Temporarily use an integer Index to sort the values, and replace by a
                 # RangeIndex in `TimeSeries.from_xarray()`
                 time_index = pd.Index(time_col_vals)
 
-            elif np.issubdtype(time_col_vals.dtype, object):
+            elif time_col_vals.dtype == nw.String:
                 # The integer conversion failed; try datetimes
                 try:
                     time_index = pd.DatetimeIndex(time_col_vals)
-                except ValueError:
+                except Exception:
                     raise_log(
                         AttributeError(
-                            "'time_col' is of 'object' dtype but doesn't contain valid timestamps"
+                            "'time_col' is of 'Utf8' dtype but doesn't contain valid timestamps"
                         )
                     )
-            elif np.issubdtype(time_col_vals.dtype, np.datetime64):
+            elif time_col_vals.dtype == nw.Datetime:
+                # BUGFIX : force time-index to be timezone naive as xarray doesn't support it
+                # pandas.DataFrame loses the tz information if it's not its index
+                if time_col_vals.dtype.time_zone is not None:
+                    logger.warning(
+                        "The provided Datetime data was associated with a timezone, which is currently not supported "
+                        "by xarray. To avoid unexpected behaviour, the tz information was removed. Consider calling "
+                        f"`ts.time_index.tz_localize({time_col_vals.dtype.time_zone})` when exporting the results."
+                        "To plot the series with the right time steps, consider setting the matplotlib.pyplot "
+                        "`rcParams['timezone']` parameter to automatically convert the time axis back to the "
+                        "original timezone."
+                    )
+                    time_col_vals = time_col_vals.dt.replace_time_zone(None)
```

Comment:
This is duplicated here (from below). We can remember the time zone here and adapt further below to simplify things (see my comment below (C1) on what you can do):

```suggestion
                # remove and remember time zone here as polars converts to UTC
                time_zone = time_col_vals.dtype.time_zone
                if time_zone is not None:
                    time_col_vals = time_col_vals.dt.replace_time_zone(None)
                time_index = pd.DatetimeIndex(time_col_vals)
```